torchvision/datasets/_stereo_matching.py

```diff
@@ -0,0 +1,1063 @@
+                        image2.right
+                        ...
+                    scene2
+                    ...
+                mixed
+                    scene1
+                        _object_settings.json
+                        _camera_settings.json
+                        image1.left.depth.png
+                        image1.right.depth.png
+                        image1.left.jpg
+                        image1.right.jpg
+                        image2.left.depth.png
+                        image2.right.depth.png
+                        image2.left.jpg
+                        image2.right
+                        ...
+                    scene2
+                    ...
+
+    Args:
+        root (string): Root directory where FallingThings is located.
+        split (string): Either "single", "mixed", or "both".
+        transforms (callable, optional): A function/transform that takes in a sample and returns a transformed version.
+
+    """
+
+    def __init__(self, root: str, split: str = "single", transforms: Optional[Callable] = None):
+        super().__init__(root, transforms)
+
+        root = Path(root) / "FallingThings"
+
+        verify_str_arg(split, "split", valid_values=("single", "mixed", "both"))
+
+        splits = {
+            "single": ["single"],
+            "mixed": ["mixed"],
+            "both": ["single", "mixed"],
+        }[split]
+
+        for s in splits:
+            imgs_left = sorted(glob(str(root / s / "*" / "*.left.jpg")))
+            imgs_right = sorted(glob(str(root / s / "*" / "*.right.jpg")))
+            if not len(imgs_left) or not len(imgs_right):
+                raise FileNotFoundError("No images found in {}".format(root))
+
+            imgs = list((left, right) for left, right in zip(imgs_left, imgs_right))
+            self._images += imgs
+
+            disparity_maps_left = sorted(glob(str(root / s / "*" / "*.left.depth.png")))
+            disparity_maps_right = sorted(glob(str(root / s / "*" / "*.right.depth.png")))
+            if not len(disparity_maps_left) or not len(disparity_maps_right):
+                raise FileNotFoundError("No disparity maps found in {}".format(root))
+
+            disparity_maps = list((left, right) for left, right in zip(disparity_maps_left, disparity_maps_right))
+            self._disparities += disparity_maps
+
+    def _read_disparity(self, file_path: str) -> Tuple:
+        # (H, W) image
+        depth = np.array(Image.open(file_path))
+        # as per https://research.nvidia.com/sites/default/files/pubs/2018-06_Falling-Things/readme_0.txt
+        # in order to extract disparity from depth maps
+        with open(os.path.split(file_path)[0] + "/_camera_settings.json", "r") as f:
+            intrinsics = json.load(f)
+            fx = intrinsics["camera_settings"][0]["intrinsic_settings"]["fx"]
+            # inverse of depth-from-disparity equation
+            disparity = (fx * 6.0 * 100) / depth.astype(np.float32)
```

Comment:
NIT could we give name to these constant `6.0` and `100` ? (by assigning a variable to them)